build fail looks real:
```
======================================================================
FAIL: test_vhnd_before_checkpoint_pregnancy (custom.opm.tests.test_windows_and_months.TestChildFirstMonthWindow)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/custom/opm/tests/test_windows_and_months.py", line 167, in test_vhnd_before_checkpoint_pregnancy
    self.assertRaises(InvalidRow, MockCaseRow, case, self.report, data_provider)
AssertionError: InvalidRow not raised
----------------------------------------------------------------------
Ran 611 tests in 57.999s
```

--- a/custom/opm/filters.py
+++ b/custom/opm/filters.py
@@ -109,15 +109,16 @@ class OpmBaseDrilldownOptionFilter(BaseDrilldownOptionFilter):
 
 
 class HierarchyFilter(OpmBaseDrilldownOptionFilter):
+    single_option_select = 0
     label = ugettext_noop("Hierarchy")
     slug = "hierarchy"
 
     @property
-    def selected(self):
-        selected = super(OpmBaseDrilldownOptionFilter, self).selected
-        if selected:
-            return selected
-        return [[ALL_OPTION]]
+    def drilldown_map(self):
+        hierarchy = super(HierarchyFilter, self).drilldown_map
+        all_hierarchy = [x for x in hierarchy
+                         if x['val'].lower() != 'all']
+        return all_hierarchy
 
 
 class MetHierarchyFilter(OpmBaseDrilldownOptionFilter):

--- a/custom/opm/reports.py
+++ b/custom/opm/reports.py
@@ -1048,6 +1048,15 @@ class NewHealthStatusReport(CaseReportMixin, BaseReport):
         return OPMCaseRow(row, self)
 
     @property
+    def fields(self):
+        return [
+            HierarchyFilter,
+            MonthFilter,
+            YearFilter,
+            OpenCloseFilter,
+        ]
+
+    @property
     def fixed_cols_spec(self):
         return dict(num=7, width=600)
 
@@ -1127,7 +1136,12 @@ class NewHealthStatusReport(CaseReportMixin, BaseReport):
     def format_cell(self, val, denom):
         if denom is None:
             return val if val is not None else ""
-        pct = " ({:.0%})".format(float(val) / denom) if denom != 0 else ""
+        if val == "NA":
+            return "NA"
+        try:
+            pct = " ({:.0%})".format(float(val) / denom) if denom != 0 else ""
+        except TypeError:
+            return "NA"
         return "{} / {}{}".format(val, denom, pct)
 
     @property

--- a/custom/opm/templates/opm/new_hsr_report.html
+++ b/custom/opm/templates/opm/new_hsr_report.html
@@ -11,10 +11,14 @@
             var totalValues= {};
             $(bodyRows).each(function(id) {
                 $(this).children().each(function(idx) {
-                    var cell;
+                    var cell = [0, 0];
                     if (idx >= begin) {
                         if (id === 0) totalValues[idx] = [0, 0];
-                        cell = $(this).text().trim().split('/').map(function(x){return parseFloat(x)});
+                        if ($(this).text().trim() != 'NA') {
+                            cell = $(this).text().trim().split('/').map(function (x) {
+                                return parseFloat(x)
+                            });
+                        }
                         if (cell.length === 2) {
                             totalValues[idx][0] += cell[0];
                             totalValues[idx][1] += cell[1];

--- a/custom/opm/tests/test_status.py
+++ b/custom/opm/tests/test_status.py
@@ -42,7 +42,9 @@ class TestPregnancyStatus(OPMCaseReportTestBase):
             forms=[],
             edd=date(2014, 3, 10),
         )
-        self.assertRaises(InvalidRow, MockCaseRow, case, self.report)
+        mock_case = MockCaseRow(case, self.report)
+        self.assertTrue(mock_case.case_is_out_of_range)
+
 
     def test_due_in_period_not_delivered(self):
         case = OPMCase(
@@ -57,4 +59,6 @@ class TestPregnancyStatus(OPMCaseReportTestBase):
             forms=[],
             edd=date(2014, 5, 10),
         )
-        self.assertRaises(InvalidRow, MockCaseRow, case, self.report)
+        mock_case = MockCaseRow(case, self.report)
+        self.assertTrue(mock_case.case_is_out_of_range)
+

--- a/custom/opm/tests/test_status.py
+++ b/custom/opm/tests/test_status.py
@@ -45,7 +45,6 @@ class TestPregnancyStatus(OPMCaseReportTestBase):
         mock_case = MockCaseRow(case, self.report)
         self.assertTrue(mock_case.case_is_out_of_range)
 
-
     def test_due_in_period_not_delivered(self):
         case = OPMCase(
             forms=[],

--- a/custom/opm/tests/test_windows_and_months.py
+++ b/custom/opm/tests/test_windows_and_months.py
@@ -39,7 +39,8 @@ class TestPregnancyWindowAndMonths(OPMCaseReportTestBase):
             forms=[],
             edd=self._offset_date(6),
         )
-        self.assertRaises(InvalidRow, MockCaseRow, case, self.report)
+        mock_case = MockCaseRow(case, self.report)
+        self.assertTrue(mock_case.case_is_out_of_range)
 
     def test_not_yet_in_range_by_a_lot(self):
         # 12 or more months in the future you also don't count
@@ -47,7 +48,8 @@ class TestPregnancyWindowAndMonths(OPMCaseReportTestBase):
             forms=[],
             edd=self._offset_date(12),
         )
-        self.assertRaises(InvalidRow, MockCaseRow, case, self.report)
+        mock_case = MockCaseRow(case, self.report)
+        self.assertTrue(mock_case.case_is_out_of_range)
 
     def test_past_range(self):
         # anytime after the period you don't count
@@ -55,14 +57,18 @@ class TestPregnancyWindowAndMonths(OPMCaseReportTestBase):
             forms=[],
             edd=self._offset_date(-1),
         )
-        self.assertRaises(InvalidRow, MockCaseRow, case, self.report)
+        mock_case = MockCaseRow(case, self.report)
+        self.assertTrue(mock_case.case_is_out_of_range)
+
 
     def test_child_first_month_not_valid(self):
         case = OPMCase(
             forms=[],
             dod=self.report_date,
         )
-        self.assertRaises(InvalidRow, MockCaseRow, case, self.report)
+        mock_case = MockCaseRow(case, self.report)
+        self.assertTrue(mock_case.case_is_out_of_range)
+
 
     def test_valid_child_month(self):
         for i in range(1, 18):
@@ -75,12 +81,13 @@ class TestPregnancyWindowAndMonths(OPMCaseReportTestBase):
             self.assertEqual(i, row.child_age)
 
     def test_child_outside_window(self):
-        with self.assertRaises(InvalidRow):
-            case = OPMCase(
-                forms=[],
-                dod=self._offset_date(-50),
-            )
-            MockCaseRow(case, self.report)
+        case = OPMCase(
+            forms=[],
+            dod=self._offset_date(-50),
+        )
+        mock_case = MockCaseRow(case, self.report)
+        self.assertTrue(mock_case.case_is_out_of_range)
+
 
 
 class TestPregnancyFirstMonthWindow(OPMCaseReportTestBase):
@@ -119,7 +126,8 @@ class TestPregnancyFirstMonthWindow(OPMCaseReportTestBase):
             edd=date(2014, 11, 15),
         )
         data_provider = MockDataProvider(default_date=date(2014, 6, 5))
-        self.assertRaises(InvalidRow, MockCaseRow, case, self.report, data_provider)
+        mock_case = MockCaseRow(case, self.report, data_provider=data_provider)
+        self.assertTrue(mock_case.case_is_out_of_range)
 
         # the next month should actually start window 4
         row = MockCaseRow(case, Report(month=7, year=2014, block="Atri"), data_provider=data_provider)
@@ -164,7 +172,8 @@ class TestChildFirstMonthWindow(OPMCaseReportTestBase):
             dod=date(2014, 5, 15),
         )
         data_provider = MockDataProvider(default_date=date(2014, 6, 5))
-        self.assertRaises(InvalidRow, MockCaseRow, case, self.report, data_provider)
+        mock_case = MockCaseRow(case, self.report, data_provider=data_provider)
+        self.assertTrue(mock_case.case_is_out_of_range)
 
         # the next month should actually start window 1
         row = MockCaseRow(case, Report(month=7, year=2014, block="Atri"), data_provider=data_provider)

